scipy/stats/_continuous_distns.py

```diff
@@ -757,37 +757,103 @@ class burr_gen(rv_continuous):
 
     .. math::
 
-        f(x, c, d) = c d x^{-c-1} (1+x^{-c})^{-d-1}
+        f(x, c, d) = c d x^{-c - 1} / (1 + x^{-c})^{d + 1}
 
     for :math:`x >= 0` and :math:`c, d > 0`.
 
     `burr` takes :math:`c` and :math:`d` as shape parameters.
 
     This is the PDF corresponding to the third CDF given in Burr's list;
-    specifically, it is equation (11) in Burr's paper [1]_.
+    specifically, it is equation (11) in Burr's paper [1]_. The distribution
+    is also commonly referred to as the Dagum distribution [2]_. If the
+    parameter :math:`c < 1` then the mean of the distribution does not
+    exist and if :math:`c < 2` the variance does not exist [2]_.
 
     %(after_notes)s
 
     References
     ----------
     .. [1] Burr, I. W. "Cumulative frequency functions", Annals of
        Mathematical Statistics, 13(2), pp 215-232 (1942).
+    .. [2] https://en.wikipedia.org/wiki/Dagum_distribution
+    .. [3] Kleiber, Christian. "A guide to the Dagum distributions."
+       Modeling Income Distributions and Lorenz Curves  pp 97-117 (2008).
 
     %(example)s
 
     """
-    _support_mask = rv_continuous._open_support_mask
+    # Do not set _support_mask to rv_continuous._open_support_mask
+    # Whether the left-hand endpoint is suitable for pdf evaluation is dependent
+    # on the values of c and d
 
     def _pdf(self, x, c, d):
         # burr.pdf(x, c, d) = c * d * x**(-c-1) * (1+x**(-c))**(-d-1)
-        return c * d * (x**(-c - 1.0)) * ((1 + x**(-c))**(-d - 1.0))
+        xx = np.asarray(x)
+        dtyp = np.find_common_type([xx.dtype, np.float64], [])
+        output = np.zeros(np.shape(xx), dtyp)
+        cond = (xx == 0)
+        if np.any(cond):
+            xz = xx[cond]
+            output[cond] = c * d * (xz**(c*d-1)) / (1 + xz**c)
+        if np.any(~cond):
+            xnz = xx[~cond]
+            output[~cond] = c * d * (xnz ** (-c - 1.0)) / ((1 + xnz ** (-c)) ** (d + 1.0))
+        if output.ndim == 0:
```

Comment:
The whole body has been reworked and simplified using a `_lazywhere`.

Reply:
I like the `_lazywhere`  change, it simplified the code some too. This might be a bit tricky for someone to comprehend though but let's leave it as your changes unless someone else also thinks is less readable.